pymc_marketing/clv/models/basic.py

```diff
@@ -78,31 +138,152 @@ def _fit_MAP(self, **kwargs):
+
+    @staticmethod
+    def _process_priors(
+        *priors: TensorVariable, check_ndim: bool = True
+    ) -> Tuple[TensorVariable, ...]:
+        """Check that each prior variable is unique and attach `str_repr` method."""
+        if len(priors) != len(set(priors)):
+            raise ValueError("Prior variables must be unique")
+        # Related to https://github.com/pymc-devs/pymc/issues/6311
+        for prior in priors:
+            if check_ndim:
+                CLVModel._check_prior_ndim(prior)
+            prior.str_repr = types.MethodType(str_for_dist, prior)  # type: ignore
+        return priors
 
     @property
-    def fit_result(self):
-        if self._fit_result is None:
+    def default_sampler_config(self) -> Dict:
+        return {}
+
+    @property
+    def _serializable_model_config(self) -> Dict:
+        return self.model_config
+
+    def sample_prior_predictive(
+        self,
+        samples: int = 1000,
+        extend_idata: bool = True,
+        combined: bool = True,
+        **kwargs,
+    ):
+        if self.model is not None:
+            with self.model:  # sample with new input data
+                prior_pred: az.InferenceData = pm.sample_prior_predictive(
+                    samples, **kwargs
+                )
+                self.set_idata_attrs(prior_pred)
+                if extend_idata:
+                    if self.idata is not None:
+                        self.idata.extend(prior_pred)
+                    else:
+                        self.idata = prior_pred
+
+        prior_predictive_samples = az.extract(
+            prior_pred, "prior_predictive", combined=combined
+        )
+
+        return prior_predictive_samples
+
+    @property
+    def prior_predictive(self) -> az.InferenceData:
+        if self.idata is None or "prior_predictive" not in self.idata:
+            raise RuntimeError(
+                "No prior predictive samples available, call sample_prior_predictive() first"
+            )
+        return self.idata["prior_predictive"]
+
+    @property
+    def fit_result(self) -> Dataset:
+        if self.idata is None or "posterior" not in self.idata:
             raise RuntimeError("The model hasn't been fit yet, call .fit() first")
-        return self._fit_result
+        return self.idata["posterior"]
 
     @fit_result.setter
-    def fit_result(self, res):
-        if self._fit_result is not None:
+    def fit_result(self, res: az.InferenceData) -> None:
+        if self.idata is None:
+            self.idata = res
+        elif "posterior" in self.idata:
             warnings.warn("Overriding pre-existing fit_result")
-        self._fit_result = res
+            self.idata.posterior = res
+        else:
+            self.idata.posterior = res
+
+    @property
+    def posterior_predictive(self) -> Dataset:
+        if self.idata is None or "posterior_predictive" not in self.idata:
+            raise RuntimeError("The model hasn't been fit yet, call .fit() first")
```

Comment:
This is wrong because `fit` doesn't trigger `sample_posterior_predictive` AFAICT. 

On a second thought I think we should remove the `prior/posterior` predictive properties and methods for now, as CLV models are a bit special in that they generally marginalize variables in a way that can't be sampled forward, or if it can, it's not really useful at all. That's why most CLV models have special models to make specific kind of posterior predictions or summaries.

Reply:
@ColtAllen do you also agree on completely removing the prior/posterior? I remember you mentioning that you had some plans considering it